services/maven-metadata/maven-metadata.service.js

```diff
@@ -69,23 +53,13 @@ export default class MavenMetadata extends BaseXmlService {
   }
 
   async handle(_namedParams, { metadataUrl, versionPrefix, versionSuffix }) {
-    const data = await this.fetch({ metadataUrl })
-    let versions = data.metadata.versioning.versions.version.reverse()
-    if (versionPrefix !== undefined) {
-      versions = versions.filter(v => v.toString().startsWith(versionPrefix))
-    }
-    if (versionSuffix !== undefined) {
-      versions = versions.filter(v => v.toString().endsWith(versionSuffix))
-    }
-    const version = versions[0]
-    // if the filter returned no results, throw a NotFound
-    if (
-      (versionPrefix !== undefined || versionSuffix !== undefined) &&
-      version === undefined
-    )
-      throw new NotFound({
-        prettyMessage: 'version prefix or suffix not found',
+    if (versionPrefix !== undefined || versionSuffix !== undefined) {
+      throw new InvalidParameter({
```

Comment:
I think it is better to explicitly throw an error here, rather than ignore the params and fall back to just showing the latest version but I could do that if we prefer.

Reply:
I agree that throwing makes things more clear. It would be next to impossible to notice a badge that doesn't quite show the right version anymore, but it's painfully obvious to see a broken badge.

Reply:
This seems quite reasonable to me. I think it's a neat way to preserve backwards compatibility.

Reply:
TODO

Reply:
https://github.com/badges/shields/pull/11077/commits/65c895287c83675a223056774c4525ede8ec8a1c